ReportUtilityBills/measure.rb

```diff
@@ -298,54 +301,42 @@ def report_runperiod_output_results(runner, utility_bills, bill_scenario_name)
-            tariff = JSON.parse(fileread, symbolize_names: true)
-            tariff = tariff[:items][0]
+          filepath = FilePath.check_path(bill_scenario.elec_tariff_filepath,
+                                         File.dirname(hpxml_path),
+                                         'Tariff File')
 
-            if tariff.keys.include?(:realtimepricing)
-              rate.fixedmonthlycharge = tariff[:fixedmonthlycharge] if tariff.keys.include?(:fixedmonthlycharge)
-              rate.realtimeprice = tariff[:realtimepricing].split(',').map { |v| Float(v) }
+          tariff = JSON.parse(File.read(filepath), symbolize_names: true)
+          tariff = tariff[:items][0]
 
-            else
-              rate.fixedmonthlycharge = tariff[:fixedmonthlycharge] if tariff.keys.include?(:fixedmonthlycharge)
-              rate.flatratebuy = tariff[:flatratebuy] if tariff.keys.include?(:flatratebuy)
+          if tariff.keys.include?(:realtimepricing)
+            rate.fixedmonthlycharge = tariff[:fixedmonthlycharge] if tariff.keys.include?(:fixedmonthlycharge)
+            rate.realtimeprice = tariff[:realtimepricing]
 
-              rate.energyratestructure = tariff[:energyratestructure] if tariff.keys.include?(:energyratestructure)
-              rate.energyweekdayschedule = tariff[:energyweekdayschedule] if tariff.keys.include?(:energyweekdayschedule)
-              rate.energyweekendschedule = tariff[:energyweekendschedule] if tariff.keys.include?(:energyweekendschedule)
+          else
+            rate.fixedmonthlycharge = tariff[:fixedmonthlycharge] if tariff.keys.include?(:fixedmonthlycharge)
+            rate.fixedmonthlycharge = tariff[:fixedchargefirstmeter] if tariff.keys.include?(:fixedchargefirstmeter)
+            rate.flatratebuy = tariff[:flatratebuy] if tariff.keys.include?(:flatratebuy)
+            rate.flatratefueladj = tariff[:flatratefueladj] if tariff.keys.include?(:flatratefueladj)
 
-              rate.demandratestructure = tariff[:demandratestructure] if tariff.keys.include?(:demandratestructure)
-              rate.demandweekdayschedule = tariff[:demandweekdayschedule] if tariff.keys.include?(:demandweekdayschedule)
-              rate.demandweekendschedule = tariff[:demandweekendschedule] if tariff.keys.include?(:demandweekendschedule)
+            rate.energyratestructure = tariff[:energyratestructure] if tariff.keys.include?(:energyratestructure)
+            rate.energyweekdayschedule = tariff[:energyweekdayschedule] if tariff.keys.include?(:energyweekdayschedule)
+            rate.energyweekendschedule = tariff[:energyweekendschedule] if tariff.keys.include?(:energyweekendschedule)
 
-              rate.flatdemandstructure = tariff[:flatdemandstructure] if tariff.keys.include?(:flatdemandstructure)
+            if tariff.keys.include?(:demandratestructure) || tariff.keys.include?(:flatdemandstructure)
+              warnings << 'Demand charges are not currently supported when calculating detailed utility bills.'
```

Comment:
Does this warning prevent the utility bills from being calculated? I would think it should since it's an incomplete calculation when demand charges are excluded.

Can you exclude JSONs from the tar.gz that have these fields?

Reply:
Yes, it does prevent the utility bill (for that scenario) from being calculated.

I _could_. Although keep in mind that we have a unit test that operates on a tariff with demand charges.

Reply:
👍 that it prevents the utility bill from being calculated and 👍 that you have a unit test.

I'd prefer that the tar.gz exclude these JSONs. It will be confusing to a BEopt user if we don't.

Reply:
I'll exclude packaging rates with any of the following fields: `demandweekdayschedule`, `demandweekendschedule`, `demandratestructure`, or `flatdemandstructure`.